tools/oeedger8r/Emitter.ml

```diff
@@ -683,6 +683,10 @@ let oe_gen_ocall_enclave_wrapper (os:out_channel) (uf:Ast.untrusted_func) =
   fprintf os "\n";
   fprintf os "{\n";
   fprintf os "    oe_result_t _result = OE_FAILURE;\n\n";
+  fprintf os "    /* If the enclave is in crashing/crashed status, new OCALL should fail\n";
+  fprintf os "       immediately. */\n";
+  fprintf os "    if (oe_get_enclave_status() != OE_OK)\n";
```

Comment:
Because this code "functioned correctly" but returned the wrong error code prior to this fix, should this status test be moved to the failure case of oe_call_host_function (line 712).  Doing so would be an optimization as this test would only be called when a failure had occurred.

Reply:
That wouldn't work. This test was returning OOM because the generated code immediately after this makes an ocall to allocate buffers. It is that ocall that fails, because _it_ sees the enclave is aborting. However, its interface, mirroring malloc, is very poor and only returns NULL as failure, thus swallowing the information that the error was due to the enclave aborting, not OOM.

Reply:
Ah, we talked offline and my understanding is that you're suggesting to optimize this by moving the comparison to where the ocall to malloc has returned NULL (to check if it was indeed an OOM or because the enclave was aborting). While reasonable, it's inconsistent with the oe_ocall implementation. We can and maybe should do this, but we should do it separately so we can optimize both implementations.

Reply:
I am fine with this implementation since it mirrors the strategy in oe_ocall where the check was always performed:
https://github.com/Microsoft/openenclave/blob/master/enclave/core/sgx/calls.c#L548

I would be up for optimizations - say we could avoid doing the host malloc if there are no arguments or for a small sized argument etc. But we can do optimizations in another pass.